File: CoffeeMaker/CoffeeMaker/CoffeeMaker.cs
using System;
namespace CoffeeMaker
{
    public class CoffeeMaker : ICoffeeMaker
    {
        private ICoffeeSelect coffeeSelect;

        public CoffeeMaker(ICoffeeSelect coffeeSelect)
        {
            this.coffeeSelect = coffeeSelect;
        }

        // Simplify
        public string MakingCoffee()
        {
            if (coffeeSelect.GetBeans() <= 0)
            {
                return "No coffee available";
            }
            if (!this.coffeeSelect.GrindBeans())
            {
                return "No ground coffee available";
            }
            return this.brew();
        }

        private string brew()
        {
            return "Coffee has now been brewed";
        }
    }
}

File: Mocking/CoffeeMaker/GoLang/CoffeeMaker.go
package CoffeeMaker

type CoffeeMaker struct {
	coffeeSelect ICoffeeSelect
}

func (coffeeMaker *CoffeeMaker) MakingCoffee() string {
	if coffeeMaker.coffeeSelect.GetBeans() <= 0 {
		return "No coffee available"
	}
	if !coffeeMaker.coffeeSelect.GrindBeans() {
		return "No ground coffee available"
	}
	return brew()
}

func brew() string {
	return "Coffee has now been brewed"
}
